ui/src/message_popup/author/new_question_page.jsx

```diff
@@ -0,0 +1,50 @@
+/* eslint no-console: "off" */
+import React from 'react';
+import _ from 'lodash';
+
+import QuestionPage from './question_page.jsx';
+import * as Api from '../../helpers/api.js';
+
+
+export default React.createClass({
+  displayName: 'NewQuestionPage',
+
+  propTypes: {
+    loaded: React.PropTypes.bool,
+    allQuestions: React.PropTypes.object,
+    onReloadQuestions: React.PropTypes.func
+  },
+
+  componentDidMount(){
+    this.props.onReloadQuestions();
+  },
+
+  getNewID(){
+    const {allQuestions} = this.props;
+    const allQuestionsFlat = allQuestions.currentQuestions.concat(allQuestions.archivedQuestions);
+    const mostRecentQuestion = _.maxBy(allQuestionsFlat, question => question.id);
+    const largestID = mostRecentQuestion !== undefined ? mostRecentQuestion.id : 0;
+    return largestID + 1;
+  },
+
+  onCreateQuestion(newQuestion){
+    const {currentQuestions, archivedQuestions} = this.props.allQuestions;
+    const question = {id: this.getNewID(), ...newQuestion};
+    const newCurrentQuestions = currentQuestions.concat(question);
+    console.log('Creating new question...');
+    console.log(question);
+    Api.saveQuestions({currentQuestions: newCurrentQuestions, archivedQuestions});
```

Comment:
I'm not too sure if I follow here. Would `onQuestionsUpdated` be replacing `onCreateQuestion`, `onEditQuestion`, and `onDeleteQuestion`? `EditQuestionPage` also needs `allQuestions` to grab the question being edited from an id, would this change?

Also, how would we handle `QuestionsPage` if we moved `QuestionsLoader` out of `app` since it needs `allQuestions` from the loader?

One thing that does make sense to me is passing a function as a prop into `QuestionsLoader` that when called updates `loaded` and `allQuestions` in `EditQuestionPage`, `NewQuestionPage`, and `QuestionsPage` but I don't think that's what you meant.